### PR TITLE
Colcon support

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -131,6 +130,7 @@ elseif(NOT EPROSIMA_INSTALLER)
 
     target_link_libraries(${PROJECT_NAME} ${PRIVACY}
         ${Boost_LIBRARIES}
+	boost::threadpool
```
Comment (MiguelCompany) [inline src/CMakeLists.txt:133]:
```suggestion
        boost::threadpool
```

File: include/transports/TCPServerTransport.h [hunk 2]
```diff
@@ -15,113 +15,125 @@
...
+
+namespace system {
+class error_code;
+} // namespace system
+} // namespace boost
+
+namespace eprosima {
+namespace rpc {
+namespace transport {
+class TCPServerTransportBoost;
```
Comment (MiguelCompany) [inline include/transports/TCPServerTransport.h:33]:
```suggestion
namespace transport {

class TCPServerTransportBoost;
```

File: rpc.repos [hunk 3]
```diff
@@ -0,0 +1,21 @@
+repositories:
+    BoostThreadpool:
+        type: git
+        url: https://github.com/richiware/boost_threadpool.git
```
Comment (MiguelCompany) [inline rpc.repos:4]: Should we fork this into our organization?

File: src/strategies/ThreadPoolStrategy.cpp [hunk 4]
```diff
@@ -15,79 +15,84 @@
 
 static const char* const CLASS_NAME = "ThreadPoolStrategy";
 
-namespace eprosima
+namespace eprosima {
+namespace rpc {
+namespace strategy {
+class ThreadPoolStrategyJob
```
Comment (MiguelCompany) [inline src/strategies/ThreadPoolStrategy.cpp:21]:
```suggestion

class ThreadPoolStrategyJob
```

File: src/strategies/ThreadPoolStrategy.cpp [hunk 5]
```diff
@@ -15,79 +15,84 @@
...
+        boost::shared_ptr<ThreadPoolStrategyJob> job(new ThreadPoolStrategyJob(callback));
+        m_pool->schedule(boost::threadpool::task_functor<void>(boost::bind(&ThreadPoolStrategyJob::run, job)));
+    }
+
+private:
+
+    boost::threadpool::pool* m_pool;
+};
+}         // namespace strategy
+}     // namespace rpc
```
Comment (MiguelCompany) [inline src/strategies/ThreadPoolStrategy.cpp:73]:
```suggestion
};

} // namespace strategy
} // namespace rpc
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/transports/TCPServerTransport.h [hunk 6]
```diff
@@ -15,113 +15,126 @@
...
+     */
+    RPC_DllAPI int receive(
+            char* buffer,
+            size_t bufferLength,
+            size_t& dataToRead,
+            Endpoint* endpoint);
+};
+
+}         // namespace transport
+}    // namespace rpc
```
Comment (MiguelCompany) [inline include/transports/TCPServerTransport.h:137]:
```suggestion
} // namespace transport
} // namespace rpc
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Sorry, I missed one

File: rpc.repos (hunk 3 above)
Comment (richiware) [reply rpc.repos:4]: &#x1F937; Original repo had errors. I've started in my personal space to check if I will able to make it work. Move this repo to eProsima is a decision to be made.

Comment (MiguelCompany) [reply rpc.repos:4]: I've gone ahead and forked it into eProsima. I think we should not have dependencies on personal repos.

Comment (richiware) [reply rpc.repos]: Done

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**